src/Chapter2.hs

```diff
@@ -620,7 +625,12 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [Int] -> [Int]
+takeEven = go []
+  where
+    go :: [Int] -> [Int] -> [Int]
+    go acc [] = acc
+    go acc (x:xs) = if even x then go (acc ++ [x]) xs else go acc xs
```

Comment:
Usually, adding elements to the end of the list is costly. Your implementation is correct :+1: But you can try to implement this function more efficiently by using `:` instead `++`. If you can think, how to avoid adding elements to the end of the list, and instead add to the beginning, but get the same result, that would be amazing!

In general, it's always a good idea to try to replace `l ++ [x]` with `x : l` whenever possible, though, in some cases it might not be that trivial to achieve the same result. Remember the analogy with trains :bullettrain_side: :train: :train: :train:

Also, your implementation takes even elements, not _elements on even positions_. Taking only even elements can be implemented simpler as `filter even` but that's not what we wanted here :slightly_smiling_face:

Reply:
I got your point. I will try to solve in efficient way. Thank you for detailed review.